### ISSUE TITLE
Starter Block still not aligning properly

### ISSUE THREAD

**Issue by Tisdag1** [tsunamayo/Starship-EVO#5690]
### Short Description of the issue:
I started a new fresh game to test this new update.
The starter block is still facing a different direction when placed than the direction it was rotated to before placing.

### Steps to reproduce the issue:

Rotate starter block to desired direction then place.

### Environment
Survival- 24w013a

### Additional information:

![20240327121648_1](https://github.com/tsunamayo/Starship-EVO/assets/95005604/c1f38421-5d61-4393-bbd5-db9ab16384dc)
This is the direction I want it facing. Directly facing me.

![20240327121658_1](https://github.com/tsunamayo/Starship-EVO/assets/95005604/ba8f58d7-6a50-4354-97d8-07bf61ab69d7)
This is the direction it ends up facing when placed.

![20240327121719_1](https://github.com/tsunamayo/Starship-EVO/assets/95005604/c67dcac9-f96a-46e2-92b2-a1a03bc4dffa)
Also, when I go to place it again in the same spot, the ghost image appears as white now instead of green. And I cannot rotate the block anymore until I move it over to a spot on the ground it wasn't placed before, where then it will turn green again. (As if there is still a block there that it wants to connect to.)

Please also include the `Player.log` file. You can find the log file here: `%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\`

[Player.log](https://github.com/tsunamayo/Starship-EVO/files/14779810/Player.log)

If you issue concerns a particular design please include any blueprint that could help to reproduce the issue.
Blueprint folder is over at:
`%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\Save_Data\Blueprints\`
You will need to zip the folder to upload it. 
To do so Right click and choose [Send to -> compressed zip file] then drag the .zip onto your report to upload it.

**Comment by tsunamayo** [comment]
@Tisdag1 Hi, I cant reproduce that issue, it places correctly. You have to remember that placing a second block might place it on the entity you previously built (even if on the terrain), this is why it appears white. You have to move a bit further away or unspawn the previous entity.
Thanks

**Comment by Tisdag1** [comment]
Hmmm... odd. I started a couple more new survival games and was still having the same problem of the block not placing the direction I rotated it to. But then I started a new creative game instead to test it there & it was working fine. (And thought, maybe it was only fixed in creative.) So for kicks and giggles I started yet another new survival game and now it's working fine there too. I made a bunch of starter blocks and they all placed in the direction I rotated them to.

Now that I think of it. The bug I was having with the hover pod issue #5678 seems like it was fixed also after I switched over to a creative game briefly. (Could there be a correlation there? Or maybe I am just imagining things. 😝 )

In any case, yeah both issues seem to be working as intended now.
Thanks again!
